lib/ansible/plugins/inventory/gcp_compute.py

```diff
@@ -322,6 +318,7 @@ def parse(self, inventory, loader, path, cache=True):
         if cache:
             try:
                 results = self.cache.get(cache_key)
+                self._add_hosts(results, config_data)
```

Comment:
Have you been able to test this? I'd expect results here to be in the format
```
{'_meta': 
    {'hostvars': {
        hostname1: {hostvar1: hostvarvalue1, hostvar2: hostvarvalue2, etc},
        hostname2: {hostvar1: hostvarvalue1, hostvar2: hostvarvalue2, etc},
    }
} 
```
which doesn't look like it's supported by _add_hosts...looks like it will try to add the host 'hostvars' and then blow up since the key 'selfLink' won't exist.

Reply:
Ah, I think I'm wrong, nevermind that comment. The cache is set a little differently than other inventory plugins I've looked at. Reviewing without being able to test it is tricky. But now looking more closely at how the cache is set I wonder if it is overwriting the cache for each zone. Sorry for not being able to actually test this out.